### PR TITLE
Add reattach flag to ECSOperator

### LINKED ISSUE #10620: Reattach ECS Task when Airflow restarts
**Description**

In similar fashion to https://github.com/apache/airflow/pull/4083, it would be helpful for Airflow to reattach itself to the ECS Task rather than letting another instance to start. However, instead of making this the default behavior, it would be better to use a `reattach` flag.

**Use case / motivation**

Allow Airflow the option to reattach to an existing ECS task when a restart happens, which would avoid having "rogue" tasks.

### PR DESCRIPTION
**Pull request by darwinyip**
closes #10620 

Adds `reattach` flag to `ECSOperator` so that whenever the Airflow server restarts, it does not leave rogue ECS Tasks. Instead the operator will seek for any running instance and attach to it.

At [Button](https://www.usebutton.com/), we use a custom ECSOperator which we added this functionality.

### CODE REVIEW

**Review by feluelle — APPROVED** [review]
LGTM in general @darwinyip 👍 

Can you fix the failing static checks, please? :)

**Review by feluelle — CHANGES_REQUESTED** [review]
Oh maybe you want to add some tests for this new logic. That would be great :)

**Comment by darwinyip** [comment]
@feluelle Can you recommend how to go about testing this?

File: tests/providers/amazon/aws/operators/test_ecs.py [hunk 1]
```diff
@@ -269,3 +269,48 @@ def test_check_success_task_not_raises(self):
...
+        client_mock.list_tasks.assert_called_once_with(
+            cluster='c', launchType=launch_type, desiredStatus='RUNNING', family='f'
+        )
+
+        start_mock.assert_not_called()
+        wait_mock.assert_called_once_with()
+        check_mock.assert_called_once_with()
+        self.assertEqual(
+            self.ecs.arn, 'arn:aws:ecs:us-east-1:012345678910:task/d8c67b3c-ac87-4ffe-a847-4785bc3a8b55'
+        )
```
Comment (feluelle) [inline tests/providers/amazon/aws/operators/test_ecs.py:316]: Exactly. That's what I had in mind. Thanks.

**Review by feluelle — APPROVED** [review]
Sorry for the late reply. It LGTM 👍

**Comment by feluelle** [comment]
Recently type annotations were added to the aws provider. That's why I would like you to rebase - and if that goes fine I am merging it.

**Pull request merged**